components/create-bounty/token-amount/create-bounty-token-amount.tsx

```diff
@@ -0,0 +1,298 @@
+
+import BigNumber from "bignumber.js";
+import { useTranslation } from "next-i18next";
+import { useDebouncedCallback } from "use-debounce";
+
+import ResponsiveWrapper from "components/responsive-wrapper";
+
+import { useAppState } from "contexts/app-state";
+
+import calculateDistributedAmounts from "helpers/calculateDistributedAmounts";
+
+import { DistributedAmounts } from "interfaces/proposal";
+
+import InputNumber from "../../input-number";
+import TokensDropdown from "../../tokens-dropdown";
+import RewardInformationBalanceView from "../reward-information/balance/view";
+import RenderItemRow from "./item-row/view";
+import ServiceFeesModalView from "./service-fees-modal/view";
+
+const ZeroNumberFormatValues = {
+  value: "",
+  formattedValue: "",
+  floatValue: 0,
+};
+
+interface DistributionsProps extends DistributedAmounts {
+  totalServiceFees: BigNumber;
+}
+
+export default function CreateBountyTokenAmount({
+  currentToken,
+  setCurrentToken,
+  addToken,
+  canAddCustomToken,
+  defaultToken = null,
+  userAddress,
+  customTokens,
+  labelSelect,
+  tokenBalance,
+  issueAmount,
+  setIssueAmount,
+  isFunders = false,
+  needValueValidation,
+  decimals = 18,
+  isFunding = false,
+}) {
+  const { t } = useTranslation(["bounty", "common", "proposal"]);
+  const [show, setShow] = useState<boolean>(false);
+  const [rewardAmount, setRewardAmount] = useState<NumberFormatValues>(ZeroNumberFormatValues);
+  const [inputError, setInputError] = useState("");
+  const [distributions, setDistributions] = useState<DistributionsProps>();
+  const {
+    state: { currentUser, Service },
+  } = useAppState();
+
+  const debouncedDistributionsUpdater = useDebouncedCallback((value, type) => handleDistributions(value, type), 500);
+
+  const amountIsGtBalance = (v: string | number, balance: BigNumber) => BigNumber(v).gt(balance)
+
+  function handleDistributions(value, type) {
+    if (!value || !Service?.network?.amounts) return;
+  
+    const { treasury, mergeCreatorFeeShare, proposerFeeShare } = Service.network.amounts;
+
+    const handleNumberFormat = (v: BigNumber) => ({
+      value: v.toFixed(),
+      floatValue: v.toNumber(),
+      formattedValue: v.toFixed()
+    })
+  
+    const initialDistributions = calculateDistributedAmounts(treasury,
+                                                             mergeCreatorFeeShare,
+                                                             proposerFeeShare,
+                                                             BigNumber(value),
+                                                        [{recipient: currentUser?.walletAddress, percentage: 100}]);
+
+    const totalServiceFees = initialDistributions
+      ? [
+        initialDistributions.mergerAmount.value,
+        initialDistributions.proposerAmount.value,
+        initialDistributions.treasuryAmount.value,
+      ].reduce((acc, value) => BigNumber(value).plus(acc),
+               BigNumber(0))
+      : BigNumber(0);
```

Comment:
There are only 3 values to sum, I think using `reduce` is a little overkill, we can just add then

Reply:
This logic is due to receiving 3 numbers as a string, so using reduce and adding to a bigNumber facilitates and reduces the complexity of the code. Since we cannot add strings and transforming the 3 variables separately would not be as simple as it is now